### ISSUE TITLE
Inconsistent critical compressibility factor?

### ISSUE THREAD

**Issue by banuti** [CoolProp/CoolProp#2163]
I am trying to get critical compressibility factors Zcr for pure fluids using CoolProp. Is there a recommended way to do that? Right now, it depends on how I calculate it, and then it's still different from tables:

Tcr = CP.PropsSI('Tcrit',fluid)
pcr = CP.PropsSI('Pcrit',fluid)
Rcr = CP.PropsSI('rhocrit',fluid)
M = CP.PropsSI('molarmass',fluid)
R = 8.3144598

Zcr1 = pcr/Rcr/R*M/Tcr
Zcr2 = CP.PropsSI('Z','T',Tcr,'P',pcr,fluid)

As examples, for water, I get 
Zcr1 = 0.22943852984823915
Zcr2 = 0.22767060631765856
, from tables (Elliott&Lira) the value is 0.233

for methanol
Zcr1 = 0.22629957151909907
Zcr2 = 0.1629708165060253
E&L: 0.224

### Versions

**CoolProp Version:** Version: 6.4.1
**Operating System and Version:** macOS 10.15.7
**Access Method:** python in jupyter notebook

Thank you!

**Comment by ibell** [comment]
A further point is that the EOS for Wagner and Pruss for water is non-analytic - you cannot evaluate it at the critical point.  So you should bump the temperature by 1e-6 or so to get a value.

**Comment by banuti** [comment]
This third option still relies on the evaluation of an EOS, would you still recommend it over Z1=pcr/Rcr/R*M/Tcr rom the critical parameters tabulated in CoolProp?

Thank you!

Daniel

**Comment by ibell** [comment]
Yes I would,  because those values are not necessarily the same as used for
the EOS.  Always safest to use the EOS.  In most cases the differences
between methods should be less than a part in 10^4 or so.  Also, make sure
you take R from the EOS.

**Comment by banuti** [comment]
Would that be through GAS_CONSTANT?

**Comment by ibell** [comment]
Yes, that's right

**Comment by banuti** [comment]
Thank you!

Daniel

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
